prime-router/src/main/kotlin/fhirengine/engine/FHIREngine.kt

```diff
@@ -0,0 +1,222 @@
+    ) {
+        /**
+         * Set the metadata instance.
+         * @return the modified workflow engine
+         */
+        fun metadata(metadata: Metadata) = apply { this.metadata = metadata }
+
+        /**
+         * Set the settings provider instance.
+         * @return the modified workflow engine
+         */
+        fun settingsProvider(settingsProvider: SettingsProvider) = apply { this.settingsProvider = settingsProvider }
+
+        /**
+         * Set the database access instance.
+         * @return the modified workflow engine
+         */
+        fun databaseAccess(databaseAccess: DatabaseAccess) = apply { this.databaseAccess = databaseAccess }
+
+        /**
+         * Set the blob access instance.
+         * @return the modified workflow engine
+         */
+        fun blobAccess(blobAccess: BlobAccess) = apply { this.blobAccess = blobAccess }
+
+        /**
+         * Set the queue access instance.
+         * @return the modified workflow engine
+         */
+        fun queueAccess(queueAccess: QueueAccess) = apply { this.queueAccess = queueAccess }
+
+        /**
+         * Build the fhir engine instance.
+         * @return the fhir engine instance
+         */
+        fun build(): FHIREngine {
+            settingsProvider = if (metadata != null) {
+                settingsProvider ?: getSettingsProvider(metadata!!)
+            } else {
+                settingsProvider ?: settingsProviderSingleton
+            }
+
+            return FHIREngine(
+                metadata ?: Metadata.getInstance(),
+                settingsProvider!!,
+                databaseAccess ?: databaseAccessSingleton,
+                blobAccess ?: BlobAccess(),
+                queueAccess ?: QueueAccess
+            )
+        }
+    }
+
+    /**
+     * Process a [message] off of the raw-elr azure queue, convert it into FHIR, and store for next step.
+     * [workflowEngine] is used for some underlying functionality that should be pulled out into a base class
+     * at some point (tech debt).
+     * [actionHistory] and [actionLogger] ensure all activities are logged
+     */
+    fun processHL7(
+        message: RawSubmission,
+        actionLogger: ActionLogger,
+        actionHistory: ActionHistory,
+        hl7Reader: HL7Reader = HL7Reader(actionLogger)
+    ) {
+        logger.trace("Processing HL7 data for FHIR conversion.")
+        try {
+            // get the hl7 from the blob store
+            val hl7messages = hl7Reader.getMessages(message.downloadContent())
+
+            if (actionLogger.hasErrors()) {
+                throw java.lang.IllegalArgumentException(actionLogger.errors.joinToString("\n") { it.detail.message })
+            }
+
+            // use fhir transcoder to turn hl7 into FHIR
+            val fhirBundles = hl7messages.map {
+                HL7toFhirTranslator.getInstance().translate(it)
+            }
+
+            logger.info("Received ${fhirBundles.size} FHIR bundles.")
```

Comment:
I know this is old, but change to debug.  Also reword to say maybe generated instead of receive?